src/main/java/me/jellysquid/mods/sodium/mixin/features/texture_tracking/MixinAnimatedTexture.java

```diff
@@ -0,0 +1,35 @@
+package me.jellysquid.mods.sodium.mixin.features.texture_tracking;
+
+import me.jellysquid.mods.sodium.client.render.texture.TextureAtlasSpriteExtended;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.List;
+
+@Mixin(TextureAtlasSprite.AnimatedTexture.class)
+public class MixinAnimatedTexture {
+    private TextureAtlasSprite parent;
```

Comment:
why not just use a shadow for `this$0`?

see `client.txt`:

```
net.minecraft.client.renderer.texture.TextureAtlasSprite$AnimatedTexture -> exm$a:
    int frame -> b
    int subFrame -> c
    java.util.List frames -> d
    int frameRowSize -> e
    net.minecraft.client.renderer.texture.TextureAtlasSprite$InterpolationData interpolationData -> f
    net.minecraft.client.renderer.texture.TextureAtlasSprite this$0 -> a
```

Reply:
I've tried, it doesn't work. It fails on launch, saying it cannot find the field.

Reply:
> why not just use a shadow for `this$0`?

> I've tried, it doesn't work. It fails on launch, saying it cannot find the field.

From experience, that's because Fabric actually gives an intermediary name to that synthetic field.
You will need to look at the bytecode in IDE to find the actual name of that field, and have a `@Dynamic` annotation on the shadow.

Reply:
That's.... concerning, but I'll try that. Thanks.

Reply:
Nope, I just checked with a bytecode editor and they're named the same in both intermediary and mojang mappings.

Reply:
Edit: Nevermind, something I did seems to have fixed it, not sure what.